src/modules/Shared/HOC/privateRoute.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import ROUTE from 'routes/constants';
+import { LOCAL_STORAGE_KEYS } from 'shared/appConstants';
+import { loadLocalStorage } from 'shared/localStorageHelpers';
+
+interface IProps {
+  Component: any;
+}
+
+const PrivateRoute: React.FC<IProps> = ({ Component }) => {
+  const auth = loadLocalStorage(LOCAL_STORAGE_KEYS.USER_AUTH);
```

Comment:
Move the data loading logic into react context to avoid localstorage computations